crates/graphql/src/schema/objects/collection.rs

```diff
@@ -0,0 +1,113 @@
+use super::prelude::*;
+
+#[derive(Debug, Clone)]
+pub struct Collection {
+    pub address: String,
+    pub name: String,
+    pub mint_address: String,
+    pub description: String,
+    pub image: Option<String>,
+}
+
+impl From<models::Nft> for Collection {
+    fn from(
+        models::Nft {
+            address,
+            name,
+            seller_fee_basis_points: _,
+            mint_address,
+            primary_sale_happened: _,
+            uri: _,
+            description,
+            image,
+            category: _,
+            model: _,
+        }: models::Nft,
+    ) -> Self {
+        Self {
+            address,
+            name,
+            mint_address,
+            description: description.unwrap_or_else(String::new),
+            image,
+        }
+    }
+}
+
+#[graphql_object(Context = AppContext)]
+impl Collection {
+    pub fn address(&self) -> &str {
+        &self.address
+    }
+
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    pub fn mint_address(&self) -> &str {
+        &self.mint_address
+    }
+
+    pub fn description(&self) -> &str {
+        &self.description
+    }
+
+    pub fn image(&self) -> Option<&str> {
+        self.image.as_deref()
+    }
+
+    pub async fn nfts(&self, ctx: &AppContext) -> FieldResult<Vec<CollectionNft>> {
+        ctx.collection_nfts_loader
+            .load(self.address.clone().into())
+            .await
+            .map_err(Into::into)
+    }
```

Comment:
Need to get limit and offset as arguments for this field or going to be returning thousands of nfts.